pkg/xds/envoy/listeners/v3/http_inbound_route_configurer.go

```diff
@@ -1,28 +1,45 @@
 package v3
 
 import (
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	envoy_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
 	envoy_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	envoy_extensions_filters_http_local_ratelimit_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/local_ratelimit/v3"
 	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	envoy_type_v3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	v3 "github.com/kumahq/kuma/pkg/xds/envoy/routes/v3"
+	"github.com/kumahq/kuma/pkg/xds/envoy/tags"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/proto"
 
 	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
 	envoy_names "github.com/kumahq/kuma/pkg/xds/envoy/names"
 	envoy_routes "github.com/kumahq/kuma/pkg/xds/envoy/routes"
 )
 
 type HttpInboundRouteConfigurer struct {
-	Service string
-	Route   envoy_common.Route
+	Service   string
+	Route     envoy_common.Route
+	RateLimit *mesh_proto.RateLimit
```

Comment:
Ok, I feel like here we are doing mix of a lot of things. The whole point of having configureres is a separation of concerns while building Envoy config. Here what I would do
 
1. `envoy_common.Route` has field `RateLimit *mesh_proto.RateLimit`. It is "higher level" object, IMHO `TypedPerFilterConfig` is just too low level at this point
2. `HttpInboundRouteConfigurer` takes `Routes`, not `Route` and it does not take `RateLimit` as an arguments
3. When calling `HttpInboundRoute` (refactor to `HttpInboundRoutes`), first - build a `Routes` based on all the policies (in our case it's only `RateLimit` so something like `func BuildInboundRoutes(rateLimit) Routes`).
4. Create a `RateLimitConfigurer` that configures `envoy_hcm.HttpConnectionManager` with `envoy.filters.http.local_ratelimit`
5. In `RoutesConfigurer` check if `Route` has `RateLimit`, if so, then build a config for this route.
   Here I'd consider refactoring RoutesConfigurer into Configurer pattern so you can configure individual elements on the Route, but we can skip it for now and do it if we were ever to configure more stuff.

The result of this is:
* `HttpInboundRoute` has 0 knowledge about RateLimit - separation of concerns
* You now have `RateLimitConfigurer` and an easy way to add RateLimit on `Route` so you can much easier reuse this on the outbound side